File: src/main/java/no/novari/flyt/integration/validation/constraints/ReferencedConfigurationExistsValidator.java
package no.novari.flyt.integration.validation.constraints;

import no.novari.flyt.integration.validation.IntegrationValidationContext;
import org.hibernate.validator.constraintvalidation.HibernateConstraintValidatorContext;

public class ReferencedConfigurationExistsValidator
        implements HibernateConstraintValidator<ReferencedConfigurationExists, Long> {

    @Override
    public boolean isValid(Long value, HibernateConstraintValidatorContext hibernateConstraintValidatorContext) {
        IntegrationValidationContext integrationValidationContext = hibernateConstraintValidatorContext
                .getConstraintValidatorPayload(IntegrationValidationContext.class);

        return value == null || integrationValidationContext.getConfiguration() != null;
    }

}

File: src/main/kotlin/no/novari/flyt/integration/validation/constraints/ReferencedConfigurationExistsValidator.kt
package no.novari.flyt.integration.validation.constraints

import no.novari.flyt.integration.validation.IntegrationValidationContext
import org.hibernate.validator.constraintvalidation.HibernateConstraintValidatorContext

class ReferencedConfigurationExistsValidator : HibernateConstraintValidator<ReferencedConfigurationExists, Long> {
    override fun isValid(
        value: Long?,
        hibernateConstraintValidatorContext: HibernateConstraintValidatorContext,
    ): Boolean {
        val integrationValidationContext =
            hibernateConstraintValidatorContext.getConstraintValidatorPayload(
                IntegrationValidationContext::class.java,
            )

        return value == null || integrationValidationContext?.configuration != null
    }
}
